sherpa/utils/__init__.py

```diff
@@ -4135,3 +4142,156 @@ def send_to_pager(txt, filename=None, clobber=False):
 
     with open(filename, 'w', encoding="UTF-8") as fh:
         print(txt, file=fh)
+
+
+# Expression terms (used to remove excess brackets from model and
+# parameter expressions).
+#
+def get_precedences_op(op: Callable) -> tuple[int, bool]:
+    """Return precedences for the operation.
+
+    Unrecognized parameters are mapped to (9, False).
+
+    Parameters
+    ----------
+    op : callable
+       The operator (e.g. np.multiply or np.power).
+
+    Returns
+    -------
+    lprec, rprec : (int, bool)
+       The left and right "precedences" (the right case only cares
+       about being set or not hence we use a boolean).
+
+    See Also
+    --------
+    get_precedence_expr, get_precedence_lhs, get_precedence_rhs
+
+    """
+
+    # We make remainder and floor_divide have the same precedence
+    # as power (aka **) just to make things clear.
+    #
+    lprec = {numpy.power: 4, numpy.remainder: 4,
+             numpy.floor_divide: 4,
+             numpy.multiply: 3, numpy.divide: 3, numpy.true_divide: 3,
+             numpy.add: 2, numpy.subtract: 2}
+
+    rprec = {numpy.power: True, numpy.remainder: True,
+             numpy.floor_divide: True}
+
```

Comment:
I suggest moving these definitions to the module level instead of keeping them inside the function. That's for two reasons:
- Performace: Every time this function is called, the dicts are assigned and then garbage collected. Put them into the module level and they will be assigned only once.
- Ability to monkey with it: Though I don't forsee it, if they are the module level, I can modify these dicts if I have some really special case (say, I use parts of sherpa with dax or jax or pytorch tensors instead of numpy arrays). I don't think we have to document that, but it's good to have the possibility to experiment with if needed: `import sherpa.utils; sherpa.utils.lprec[pytorch.operator.power] = 4`

Reply:
And then say in the docs: "This function uses the module-level dictionaries lprec and rprec."

Reply:
Maybe there should be a `sherpa.utils.op` module to place this in (name open to bike-shedding), as that makes the scope of such module-level globals a bit-more manageable.

Reply:
That is a good idea!

Reply:
This is now in `sherpa.models.op` and they are module-level values, referenced from the documentation.